arch/risc-v/src/common/supervisor/riscv_sbi.c

```diff
@@ -54,5 +64,9 @@ void riscv_sbi_ack_timer(void)
 
 uintptr_t riscv_sbi_mhartid(void)
 {
+#ifdef CONFIG_NUTTSBI
+  return riscv_mcall_mhartid();
```

Comment:
The spec. does not provide this either, which I don't really understand why.

Knowing the current execution hart can be useful, e.g. when acking external interrrupts. A hart can claim the interrupt, but it must know which hart it is to do this.

Reply:
> The spec. does not provide this either, which I don't really understand why.
> 

Me too. It's very important to get hartid as fast as possible. 

> Knowing the current execution hart can be useful, e.g. when acking external interrrupts. A hart can claim the interrupt, but it must know which hart it is to do this.

https://github.com/riscv-non-isa/riscv-sbi-doc/issues/25
look like we can save hartid to sscratch CSR.

Reply:
Other info that could be stored into sscratch is the hart specific interrupt stack, so it does not need to be calculated every time an exception runs